tracking/src/main/java/org/hps/recon/tracking/TrackHitKiller.java

```diff
@@ -0,0 +1,555 @@
+            int layer = -1;
+            boolean top = false;
+            boolean stereo = false;
+            boolean slot = false;
+            System.out.println("Parsing ratioFile = " + ratioFile);
+            String[] tokens = ratioFile.split(delims);
+            Matcher m = layerPattern.matcher(tokens[1]);
+            if (m.find()) {
+                layer = Integer.parseInt(m.group(1));
+                if (m.group(2).matches("t"))
+                    top = true;
+            } else {
+                System.out.println("Couldn't find layer number!!!  " + ratioFile);
+                continue;
+            }
+//            if (tokens[2].matches("top"))
+//                top = true;
+//            System.out.println(tokens[2]+" "+tokens[3]+" "+tokens[4]);
+            if (tokens[2].matches("stereo"))
+                stereo = true;
+            if (tokens[3].matches("slot"))
+                slot = true;
+            System.out.println("TrackHitKiller::Killing this:  "
+                    + "layer = " + layer + "; top = " + top + "; stereo = " + stereo
+                    + "; slot = " + slot);
+            this.registerSensor(layer, top, stereo, slot, ratioFile);            
+        }
+        System.out.println("filling smearing fractions");
+        _smearingFractionsL1 = this.fillSmearedFractions(sigmaUL1, 1);
+        _smearingFractionsL6 = this.fillSmearedFractions(sigmaUL6, 12);
+    }
+
+    @Override
+    public void process(EventHeader event) {
+        //    System.out.println("In process of SVTHitKiller");
+        //        RelationalTable hitToStrips = TrackUtils.getHitToStripsTable(event);
+        
+        if (event.hasItem(stripHitInputCollectionName))
+            siClusters = (List<TrackerHit>) event.get(stripHitInputCollectionName);
+        else {
+            System.out.println("StripHitKiller::process No Input Collection Found?? " + stripHitInputCollectionName);
+            return;
+        }
+        
+        if (!event.hasCollection(LCRelation.class, helicalTrackHitRelationsCollectionName) || !event.hasCollection(LCRelation.class, rotatedHelicalTrackHitRelationsCollectionName))
+            return;
+        RelationalTable hitToStrips = TrackUtils.getHitToStripsTable(event);
+        RelationalTable hitToRotated = TrackUtils.getHitToRotatedTable(event);
+
+        List<Track> tracks = event.get(Track.class, trackCollectionName);
+
+        List<TrackerHit> l1HitsOnTracks=getTrackHitsPerLayer(tracks,hitToStrips,hitToRotated,1);
+        List<TrackerHit> tmpClusterList=new ArrayList<TrackerHit>(siClusters);
+
+        int oldClusterListSize = siClusters.size();
+            //for (TrackerHit siCluster : siClusters) {
+        for (TrackerHit siCluster : l1HitsOnTracks) {
+            HpsSiSensor sensor = (HpsSiSensor) ((RawTrackerHit) siCluster.getRawHits().get(0)).getDetectorElement();            
+            for (SensorToKill sensorToKill : _sensorsToKill)
```

Comment:
is this missing the {} brackets around the part that we should loop over?

Reply:
I guess it also works without because it's just looping over the if statement? anyway, adding the brackets would make it easier to read